server/bin/pbench-sync-satellite.sh

```diff
@@ -250,10 +250,30 @@ for host in $hosts ;do
         for x in $state_list; do
             echo "$host/TO-SYNC/$x" >> ${state_change_log}
         done
+        rm $logdir/$host/ok-checks.log
     fi
+
+    if [[ ! -s $logdir/$host/fail-checks.log ]] ; then
+        rm $logdir/$host/fail-checks.log
+    fi
+
+    if [[ -z $(grep -E -v 'OK|FAILED' $logdir/$host/md5-checks.log) ]] ; then
```

Comment:
If processing all the tar balls for this `${host}` is successful, then `fail-checks.log` will be empty because [line 217](https://github.com/distributed-system-analysis/pbench/pull/1354/files#diff-2b119fd9bfaa54cda0c0fbdad8ae2257R217) did not find any lines with "`FAILED`" in the `md5-checks.log` file.  Given that logic, then the only time [line 260](https://github.com/distributed-system-analysis/pbench/pull/1354/files#diff-2b119fd9bfaa54cda0c0fbdad8ae2257R260) will evaluate to `True` is when the `md5sum` command on [line 200](https://github.com/distributed-system-analysis/pbench/pull/1354/files#diff-2b119fd9bfaa54cda0c0fbdad8ae2257R200) emits text lines that do not have "`OK`" or "`FAILED`" on the line.

Rather than have a separate check, since the only time we would want to remove that file is if there are no "`FAILED`" checks, can we consider moving this logic to the same `then` clause of [line 257](https://github.com/distributed-system-analysis/pbench/pull/1354/files#diff-2b119fd9bfaa54cda0c0fbdad8ae2257R257)? .

Reply:
The check here is too complicated: run grep and get its output and then check if the output is the empty string. `grep` has an option for checking if there is a match or not, without producing any output: `grep -q -E -v 'OK|FAIL' <somefile>` returns success if `<somefile>` contains something other than OK or FAIL, failure otherwise. So the test above can be written more simply and more efficiently like this:

    if grep -q -E -v 'OK|FAIL' $logdir/$host/md5-checks.log ;then
         ....

That returns as soon as the answer is known instead of going through the whole file, it avoids having to gather the output of `grep` intro a string in order to check whether it is the empty string and it avoids the explicit test. But the main thing it achieves is simplicity and readability.

Reply:
@ndokos it Exits immediately with zero status if any match is found, even if an error was detected. So, it is not possible to use -q option.